bulk_grades/api.py

```diff
@@ -224,12 +226,12 @@ def _get_graded_subsections(course_id, filter_subsection=None, filter_assignment
         subsections = OrderedDict()
         for subsection in grades_api.graded_subsections_for_course_id(course_id):
             block_id = text_type(subsection.location.block_id)
-            if (
+            if (  # pragma: no branch
                     (filter_subsection and (block_id != filter_subsection.block_id))
                     or
                     (filter_assignment_type and (filter_assignment_type != text_type(subsection.format)))
             ):
-                continue
+                continue  # pragma: no cover
```

Comment:
What is the reason for explicit no cover/no branch statements (i.e: platform doesn't seem to be using it).

Reply:
Yeah, so I really am not sure what's happening here. I have tests that test this code (https://github.com/edx/edx-bulk-grades/pull/37/files?file-filters%5B%5D=.py#diff-625efbbf35a2290861f596fee1b19838R197) but this still shows up as red. Even when stepping through with pdb, when line 230 is true, the debugger never goes to the continue statement, it jumps back to the for loop.

I wanted to explicitly cover them so that the coverage file matched reality, despite the weirdness from the coverage tool

Reply:
looks like yeah, this is a quirk with some under-the-hood optimization
https://bitbucket.org/ned/coveragepy/issues/198/continue-marked-as-not-covered